scripts/sanitize.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+set -eu
+set -o pipefail
+
+: '
+
+Rebuilds the code with the sanitizers and runs the tests
+
+'
+
+# See https://github.com/mapbox/node-cpp-skel/blob/master/docs/extended-tour.md#configuration-files
+
+make clean
+
+# https://github.com/google/sanitizers/wiki/AddressSanitizerAsDso
+SHARED_LIB_EXT=.so
+if [[ $(uname -s) == 'Darwin' ]]; then
+    SHARED_LIB_EXT=.dylib
+fi
+
+if [[ $(uname -s) == 'Linux' ]]; then
+    ./node_modules/.bin/mason-js install binutils=2.30 --type=compiled
+    ./node_modules/.bin/mason-js link binutils=2.30 --type=compiled
+    export PATH=$(pwd)/mason_packages/.link/bin:${PATH}
+fi
+export MASON_LLVM_RT_PRELOAD=$(pwd)/$(ls mason_packages/.link/lib/clang/*/lib/*/libclang_rt.asan*${SHARED_LIB_EXT})
+SUPPRESSION_FILE="/tmp/leak_suppressions.txt"
+echo "leak:__strdup" > ${SUPPRESSION_FILE}
+echo "leak:v8::internal" >> ${SUPPRESSION_FILE}
+echo "leak:node::CreateEnvironment" >> ${SUPPRESSION_FILE}
+echo "leak:node::Init" >> ${SUPPRESSION_FILE}
```

Comment:
Why these suppressions? And should this be a suppression file for this repo and not implicitly generated behind the user's back?